hoomd/md/pair.py

```diff
@@ -1641,127 +1647,29 @@ def __init__(self, nlist, r_cut=None, r_on=0., mode='none'):
         self._add_typeparam(params)
 
 
-class _shape_dict(dict):
-    """Simple dictionary subclass to improve handling of anisotropic potential
-    shape information."""
-    def __getitem__(self, key):
-        try:
-            return super(_shape_dict, self).__getitem__(key)
-        except KeyError as e:
-            raise KeyError("No shape parameters specified for particle type {}!".format(key)) from e
-
-
-class ai_pair(pair):
+class _AnisotropicPair(Pair):
     R"""Generic anisotropic pair potential.
 
-    Users should not instantiate :py:class:`ai_pair` directly. It is a base class that
-    provides common features to all anisotropic pair forces. Rather than repeating all of that documentation in a
-    dozen different places, it is collected here.
-
-    All anisotropic pair potential commands specify that a given potential energy, force and torque be computed
-    on all non-excluded particle pairs in the system within a short range cutoff distance :math:`r_{\mathrm{cut}}`.
-    The interaction energy, forces and torque depend on the inter-particle separation
-    :math:`\vec r` and on the orientations :math:`\vec q_i`, :math:`q_j`, of the particles.
+    Users should not instantiate :py:class:`ai_pair` directly. It is a base
```

Comment:
I was thinking that there were no specific details of `_AnisotropicPair` that needed to be exposed. I don't think they should be their own namespace. We could consider a submodule `md.pair.aniso.Dipole`.

Reply:
I was thinking we could use it to show that anisotropic potentials  all have `r_cut` and the shift modes like we do for `Pair`.

I like the idea of using `md.pair.aniso` for the potentials. It would make them stand out in the documentation and make it clear that they are different from the isotropic pair potentials. Let's do that.